core/src/main/scala/kafka/server/DelayedOperation.scala

```diff
@@ -292,54 +286,47 @@ class DelayedOperationPurgatory[T <: DelayedOperation](purgatoryName: String,
    */
   private class Watchers(val key: Any) {
 
-    private[this] val operations = new LinkedList[T]()
+    private[this] var operations = List[T]()
```

Comment:
I was thinking about this also. I suppose it would be fine if a `tryCompleteWatched` or a `purgeCompleted` missed an update in `watch`? But yikes, I'm surprised to see O(n) for `size`. Was that to save space?